scripts/subdetector_tests/g4_material_scan.py

```diff
@@ -0,0 +1,147 @@
+    compact: is path to compact file
+    start_point: a list or tuple for 3D coordinates (e.g., 0,0,0)
+    direction: a lit or tuple for direction (e.g., 0.1,0.2,1.0)
+    return: a dataframe for materialScan results
+'''
+def g4_material_scan(compact, start_point, direction, timeout=200):
+    EXEC_NAME = 'g4MaterialScan'
+    cmd = '{} --compact={} --position=\"{},{},{}\" --direction=\"{},{},{}\"'\
+          .format(EXEC_NAME, compact, *np.hstack([start_point, direction]))
+    output = os.popen(cmd).read()
+    # output = subprocess.check_output(cmd.split(' '), timeout=timeout).decode('UTF-8')
+
+    # find material scan lines
+    lines = []
+    add_line = False
+
+    for l in output.split('\n'):
+        if add_line:
+            lines.append(l.strip())
+        if l.strip().startswith('MaterialScan'):
+            add_line = True
+
+
+    # NOTE: the code below is for materialScan output as of 03/05/2023
+    # it may need change if the output format is changed
+    scans = []
+    first_digit = False
+    for i, l in enumerate(lines):
+        line = l.strip('| ')
+        if not first_digit and not line[:1].isdigit():
+            continue
+        first_digit = True
+        if not line[:1].isdigit():
+            break
+        # break coordinates for endpoint, which has a format of (x, y, z)
+        scans.append(line.strip('| ').translate({ord(i): None for i in '()'}).replace(',', ' '))
+
+    cols = [
+        'material', 'Z', 'A', 'density',
+        'rad_length', 'int_length', 'thickness', 'path_length',
+        'int_X0', 'int_lamda', 'end_x', 'end_y', 'end_z'
+        ]
+
+    dft = pd.read_csv(StringIO('\n'.join(scans)), sep='\s+', header=None, index_col=0, names=cols)
+    return dft.astype({key: np.float64 for key in cols[1:]})
+
+
+'''
+    A helper function to convert a string (<min>[:<max>][:<step>]) to an array
+'''
+def args_array(arg, step=1, include_end=True):
+    vals = [float(x.strip()) for x in arg.split(':')]
+    # empty or only one value
+    if len(vals) < 2:
+        return np.array(vals)
+    # has step input
+    if len(vals) > 2:
+        step = vals[2]
+    # inclusion of the endpoint (max)
+    if include_end:
+        vals[1] += step
+    return np.arange(vals[0], vals[1], step)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+            dest='compact',
+            help='Top-level xml file of the detector description.'
+            )
+    parser.add_argument(
+            '-o', '--output', default='g4_materials.csv',
+            help='A csv file contains the scan info.'
+            )
+    parser.add_argument(
+            '--start-point', default='0,0,0',
+            help='Start point of the scan, use the format \"x,y,z\", unit is cm.'
+            )
+    parser.add_argument(
+            '--eta', default='-4.0:4.0:0.1',
+            help='Eta range, in the format of \"<min>[:<max>][:<step>]\".'
```

Comment:
(minor) 
Technically it's: 
```suggestion
            help='Eta range, in the format of \"<min>[:<max>[:<step>]]\".'
```

Reply:
Yes, you're right